compiler/sem/semtypes.nim

```diff
@@ -2124,7 +2112,9 @@ proc semTypeNode(c: PContext, n: PNode, prev: PType): PType =
             # I suggest revisiting this once the language decides on whether
             # `not nil` should be the default. We can then map nilable refs
             # to other types such as `Option[T]`.
-            result = makeTypeFromExpr(c, newTree(nkStmtListType, n.copyTree))
+            var copyN = n.copyTree
+            copyN[1].typ = makeTypeDesc(c, result)
```

Comment:
```suggestion
            # the `not` procedure expects a typedesc, so make sure the
            # operand is a `tyTypeDesc`  
            copyN[1].typ = makeTypeDesc(c, result)
```
A soft suggestion that would improve clarity a bit, I think. Feel free to ignore.